lib/HTTP/Tiny.pm

```diff
@@ -1644,12 +1644,12 @@ sub _find_CA_file {
 
     my $ca_file =
       defined( $self->{SSL_options}->{SSL_ca_file} )
-      ? $self->{SSL_options}->{SSL_ca_file}
-      : $ENV{SSL_CERT_FILE};
+      ? [ 'SSL_options->{SSL_ca_file}', $self->{SSL_options}->{SSL_ca_file} ]
+      : [ 'SSL_CERT_FILE', $ENV{SSL_CERT_FILE} ];
 
-    if ( defined $ca_file ) {
-        unless ( -r $ca_file ) {
-            die qq/SSL_ca_file '$ca_file' not found or not readable\n/;
+    if ( defined $ca_file->[1] ) {
+        unless ( -r $ca_file->[1] ) {
+            die qq/'$ca_file' from $ca_file->[0] not found or not readable\n/;
```

Comment:
I'd prefer either using to scalar variables for the CA file source for the message and its path+filename OR a hashref with descriptive key names, e.g. ca_file_source and ca_file.